cpp/arcticdb/entity/data_error.hpp

```diff
@@ -25,30 +25,32 @@ enum class VersionRequestType: uint32_t {
 class DataError {
 public:
     DataError(const StreamId& symbol,
-              const pipelines::VersionQueryType& version_query_type,
+              const std::optional<pipelines::VersionQueryType>& version_query_type,
```

Comment:
It is much cleaner to reorder the arguments so that `version_query_type` is defaulted to `std::nullopt`, rather than callers having to explicitly provide `std::nullopt`

Reply:
yes it is cleaner, agree